### PR TITLE
Add option to fully reload weapons that reload single rounds

### PR DESCRIPTION
**Pull request by WelpThatWorked**
#### Summary
Interface "Allow full reloads of weapons that reload one round at a time with one action. Interrupted full reloads will partially reload the weapon."

#### Purpose of change

Allow weapons with flag RELOAD_ONCE to be reloaded fully without having to hit reload multiple times. 

#### Describe the solution

Make a new customized inventory selector based off of the pick selector with a selector for "Reload Once / Full Reload", as well as the functions needed for it to be used by reload_item(). 
Add inv menu item for RELOAD_ONCE weapons to do a full reload
Modify reload(), reload_option and the reload activity to allow for proper full reloads, as well as interrupted reloads of RELOAD_ONCE weapons.

#### Describe alternatives you've considered

A menu that appears _after_ selecting weapon and ammo, asking if you want to do a full reload. Would require more keypresses for normal use. 
Making full reloads the default option
Alternatives to the descriptor "full reload". Unsure if this is clear enough to what the option will do. 

#### Testing

Tested, with M1903, single reloads, full reloads, reloads with speedloader, interrupted full reloads. Returned as expected. 

#### Additional context

image
image

### CODE REVIEW

File: src/activity_actor.cpp [hunk 1]
```diff
@@ -2679,8 +2679,67 @@ void reload_activity_actor::finish( player_activity &act, Character &who )
...
+            reloadable.set_flag( flag_FILTHY );
+        }
+
+        if( reloadable.get_var( "dirt", 0 ) > 7800 ) {
+            add_msg( m_neutral, _( "You manage to loosen some debris and make your %s somewhat operational." ),
+                     reloadable_name );
+            reloadable.set_var( "dirt", ( reloadable.get_var( "dirt", 0 ) - rng( 790, 2750 ) ) );
+        }
+
+        add_msg( m_neutral, _( "You insert %dx %s into the %s." ), num_loaded, ammo_name, reloadable_name );
```
Comment (actual-nh) [inline src/activity_actor.cpp:2715]:
```suggestion
        add_msg( m_neutral, _( "You insert %1$dx %2$s into the %3$s." ), num_loaded, ammo_name, reloadable_name );
```
These are positional parameters to help the translators - another language may not order the words the same way.

File: src/player.h [hunk 2]
```diff
@@ -223,15 +223,15 @@ class player : public Character
         int get_lift_assist() const;
 
         bool list_ammo( const item &base, std::vector<item::reload_option> &ammo_list,
-                        bool empty = true ) const;
+                        bool empty = true, bool fullReload = false ) const;
```
Comment (anothersimulacrum) [inline src/player.h:226]: For the future (no need to change here), `snake_case` is the preferred style in this codebase.

Comment (WelpThatWorked) [reply src/player.h:226]: got it

**Comment by LaVeyanFiend** [comment]
If you're interrupted on a full reload, does it undo the entire reload as is current or give you a partial reloading?

**Comment by WelpThatWorked** [comment]
Yes, as in this screenshot, if you are interrupted by a distraction while doing a full reload of a weapon that reloads single rounds, you will get a number of rounds that would have been fully inserted in the time elapsed. 
image

**Comment by TealcOneill** [comment]
@WelpThatWorked Would you like me to poke the devs about the failed tests on the discord?

**Comment by I-am-Erk** [comment]
I am closing this PR as abandoned for now, please comment if you'd like to return and finish it off.

**Pull request closed without merging**